### PR TITLE
DT-727 Add a functional test

### PR DESCRIPTION
**Pull request by mikehalmamoj**
And switch unit test to normal testing stack

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -36,7 +36,9 @@ dependencies {
 
   implementation("org.springframework.boot:spring-boot-gradle-plugin:2.2.6.RELEASE")
 
-  testImplementation("io.kotlintest:kotlintest-runner-junit5:3.3.1")
+  testImplementation("org.junit.jupiter:junit-jupiter:5.5.2")
+  testImplementation("org.mockito:mockito-junit-jupiter:3.1.0")
+  testImplementation("org.assertj:assertj-core:3.13.2")
```
Comment (petergphillips) [inline build.gradle.kts:41]: old version?  Worth grabbing the `useLatestVersions` plugin :grin:

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginFuncTest.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  // TODO DT-727 Next task is to move everything below the plugins section to the plugin and make sure that these tests still pass
+  private fun makeBuildScript() {
+    val buildFile = File(projectDir, "build.gradle.kts")
+    val buildScript = """
+        import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+  
+        plugins {
+          kotlin("jvm") version "1.3.71"
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginFuncTest.kt:93]: old version?  How do we keep these up to date?

File: build.gradle.kts (hunk 1 above)
Comment (mikehalmamoj) [reply build.gradle.kts:41]: Cool, will do in next PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginFuncTest.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginFuncTest.kt:93]: Yeah great question.
Don't currently have an answer - but I will look into it.

Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/DpsSpringBootPluginFuncTest.kt:93]: I have an answer - the version of this is set by the gradle plugin's build script (in a call to dependencies/implementation).  Therefore we run the `useLatestVersions` task against the plugin.

**Pull request merged**